notebook/services/contents/largefilemanager.py

```diff
@@ -0,0 +1,73 @@
+from notebook.services.contents.filemanager import FileContentsManager
+from contextlib import contextmanager
+from tornado import web
+import nbformat
+import base64
+import os, io
+
+class LargeFileManager(FileContentsManager):
+    """Handle large file upload."""
+
+    def save(self, model, path=''):
+        """Save the file model and return the model with no content."""
+        chunk = model.get('chunk', None)
+        if chunk is not None:
+            path = path.strip('/')
+            
+            if 'type' not in model:
+                raise web.HTTPError(400, u'No file type provided')
+            if model['type'] != 'file':
+                raise web.HTTPError(400, u'File type "{}" is not supported for large file transfer'.format(model['type']))
+            if 'content' not in model and model['type'] != 'directory':
+                raise web.HTTPError(400, u'No file content provided')
+
+            os_path = self._get_os_path(path)
+
+            # First chunk
+            if chunk == 1:
+                self.log.debug("Saving %s", os_path)
+                self.run_pre_save_hook(model=model, path=path)
+            try:
+                # First chunk
+                if chunk == 1:
+                    super(LargeFileManager, self)._save_file(os_path, model['content'], model.get('format'))
```

Comment:
Is there a reason not to merge the `if chunk == 1` block here with the one 5 lines above ?